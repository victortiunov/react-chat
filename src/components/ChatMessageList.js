import React from 'react';
import { withStyles } from 'material-ui/styles';
import ChatMessage from './ChatMessage';

const styles = theme => ({
	messagesWrapper: {
		overflowY: 'scroll',
		height: 'calc(100% - 60px)',
		width: '100%',
		paddingTop: theme.spacing.unit * 3,
		marginBottom: '70px',
	},
});

class ChatMessageList extends React.Component {
	componentDidMount() {
		this.scrollDownHistory();
	}

	componentDidUpdate() {
		this.scrollDownHistory();
	}

	scrollDownHistory() {
		if (this.messagesWrapper) {
			this.messagesWrapper.scrollTop = this.messagesWrapper.scrollHeight;
		}
	}

	render() {
		const { classes, user, messages } = this.props;

		return (
			<div
				className={classes.messagesWrapper}
				ref={(wrapper) => { this.messagesWrapper = wrapper; }}
			>
				{messages && messages.map(message => (
					// eslint-disable-next-line
					<ChatMessage key={message._id} user={user} {...message} />
				))}
			</div>
		);
	}
}

export default withStyles(styles)(ChatMessageList);
