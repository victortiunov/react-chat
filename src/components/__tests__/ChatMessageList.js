/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatMessageList from '../ChatMessageList';

// Return a fixed timestamp when moment().fromNow() is called
jest.mock('moment', () => () => ({ fromNow: () => '37 days ago' }));

const mockProps = {
  messages: [
    {
      _id: '73to37',
      chatId: '37gt37',
      content: ' joined',
      sender: {
        _id: '737373',
        username: 'dobby',
        firstName: 'Dobby',
        lastName: 'Elf',
      },
      createdAt: '1998-03-26T10:53:23.200Z',
    },
    {
      _id: '212121',
      chatId: '37gt37',
      content: 'Dobby!!!',
      sender: {
        _id: '111111',
        username: 'hgranger',
        firstName: 'Hermione',
        lastName: 'Granger',
      },
      createdAt: '1998-03-26T10:53:23.200Z',
    },
    {
      _id: '377337',
      chatId: '37gt37',
      content: 'Hi, Dobby!',
      sender: {
        _id: '373737',
        username: 'hpotter',
        firstName: 'Harry',
        lastName: 'Potter',
      },
      createdAt: '1998-03-26T10:53:23.200Z',
    },
  ],
  user: {
    _id: '373737',
  },
};

describe('<ChatMessageList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatMessageList {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ChatMessageList {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without messages', () => {
    const tree = renderer.create(<ChatMessageList {...mockProps} messages={[]} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
