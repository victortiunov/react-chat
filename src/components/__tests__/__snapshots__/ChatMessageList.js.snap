// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMessageList /> renders correctly 1`] = `
<div
  className="ChatMessageList-messagesWrapper-62"
>
  <div
    className="ChatMessage-messageWrapper-63"
  >
    <div
      className="MuiAvatar-root-69 MuiAvatar-colorDefault-70"
      style={
        Object {
          "backgroundColor": "#e91e63",
        }
      }
    >
      DE
    </div>
    <div
      className="MuiPaper-root-72 MuiPaper-elevation2-76 MuiPaper-rounded-73 ChatMessage-message-65"
    >
      <span
        className="MuiTypography-root-99 MuiTypography-caption-109"
      >
        Dobby Elf
      </span>
      <p
        className="MuiTypography-root-99 MuiTypography-body1-108"
      >
         joined
        <span
          className="MuiTypography-root-99 MuiTypography-caption-109"
        >
          37 days ago
        </span>
      </p>
    </div>
  </div>
  <div
    className="ChatMessage-messageWrapper-63"
  >
    <div
      className="MuiAvatar-root-69 MuiAvatar-colorDefault-70"
      style={
        Object {
          "backgroundColor": "#009688",
        }
      }
    >
      HG
    </div>
    <div
      className="MuiPaper-root-72 MuiPaper-elevation2-76 MuiPaper-rounded-73 ChatMessage-message-65"
    >
      <span
        className="MuiTypography-root-99 MuiTypography-caption-109"
      >
        Hermione Granger
      </span>
      <p
        className="MuiTypography-root-99 MuiTypography-body1-108"
      >
        Dobby!!!
        <span
          className="MuiTypography-root-99 MuiTypography-caption-109"
        >
          37 days ago
        </span>
      </p>
    </div>
  </div>
  <div
    className="ChatMessage-messageWrapper-63 ChatMessage-messageWrappperFromMe-64"
  >
    <div
      className="MuiPaper-root-72 MuiPaper-elevation2-76 MuiPaper-rounded-73 ChatMessage-message-65 ChatMessage-messageFromMe-66"
    >
      <span
        className="MuiTypography-root-99 MuiTypography-caption-109"
      >
        Harry Potter
      </span>
      <p
        className="MuiTypography-root-99 MuiTypography-body1-108"
      >
        Hi, Dobby!
        <span
          className="MuiTypography-root-99 MuiTypography-caption-109"
        >
          37 days ago
        </span>
      </p>
    </div>
    <div
      className="MuiAvatar-root-69 MuiAvatar-colorDefault-70"
      style={
        Object {
          "backgroundColor": "#009688",
        }
      }
    >
      HP
    </div>
  </div>
</div>
`;

exports[`<ChatMessageList /> renders correctly without messages 1`] = `
<div
  className="ChatMessageList-messagesWrapper-62"
/>
`;
